src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembers/DynamicallyAccessedMembersAnalysis.cs

```diff
@@ -0,0 +1,85 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using ILLink.Shared;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.FlowAnalysis;
+
+namespace ILLink.RoslynAnalyzer
+{
+	public readonly struct BlockWrapper : IEquatable<BlockWrapper>
+	{
+		public readonly BasicBlock Block;
+
+		public BlockWrapper (BasicBlock block) => Block = block;
+
+		public IEnumerable<BlockWrapper> Predecessors {
+			get {
+				foreach (var predecessor in Block.Predecessors)
+					yield return new BlockWrapper (predecessor.Source);
+			}
+		}
+
+		public bool Equals (BlockWrapper other) => Block.Equals (other.Block);
+	}
+
+	public readonly struct ControlFlowGraphWrapper : IControlFlowGraph<BlockWrapper>
+	{
+		readonly ControlFlowGraph ControlFlowGraph;
+
+		public ControlFlowGraphWrapper (ControlFlowGraph cfg) => ControlFlowGraph = cfg;
+
+		public IEnumerable<BlockWrapper> Blocks {
+			get {
+				foreach (var block in ControlFlowGraph.Blocks)
+					yield return new BlockWrapper (block);
+			}
+		}
+
+		public BlockWrapper Entry => new BlockWrapper (ControlFlowGraph.Blocks[0]);
+
+		public IEnumerable<BlockWrapper> GetPredecessors (BlockWrapper block)
+		{
+			foreach (var predecessor in block.Block.Predecessors)
+				yield return new (predecessor.Source);
+		}
+	}
+
+	public class DynamicallyAccessedMembersAnalysis
```

Comment:
TrimAnalysis maybe?

Reply:
I went with TrimDataFlowAnalysis since it's only one part of the trim analysis.